### ISSUE TITLE
Flase positive of 2059 and 2005 when combining printf and gettext

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1098]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/
SIGSTRENGTH="10"
printf "$(gettext "%d%%")" "$((SIGSTRENGTH))"
echo "$(gettext "Unknown")"

```

#### Here's what shellcheck currently says:
Gives 2059 and 2005.

#### Here's what I wanted or expected to see:
Nothing, since this is the way to use 'gettext' IMHO.

**Comment by deryni** [comment]
About SC2005, if you have other contents in the string passed to echo then it shouldn't trigger and if you *don't* have other contents in the string passed to echo then why is SC2005 not correct? Do you need to use `echo "$(gettext "Unknown")"` instead of just `gettext "Unknown"` for some reason there to get the output you expect? (Though perhaps you need `gettext -s "Unknown"` there instead?)

**Comment by nihilus** [comment]
IIRC gettext -s behaves like echo -n, which is not what I want.
